### ISSUE TITLE
Latest SNP OVMF binary causes measurement failure

### ISSUE THREAD

**Issue by ShannonSD** [virtee/sev-snp-measure#50]
The OVMF binary that's built using the AMDSEV repository now contains a SVSM Calling Area Address (CAA) section type.
Defined [here](https://github.com/AMDESE/ovmf/blob/snp-latest/OvmfPkg/ResetVector/X64/OvmfSevMetadata.asm) as:
```
%define OVMF_SECTION_TYPE_SVSM_CAA        0x4
```
This causes the measurement to fail with 
```
ValueError: 4 is not a valid SectionType
``` 

Could you give me some guidance on how I could fix the measurement calculation?

**Comment by dubek** [comment]
Thank you @ShannonSD for reporting this issue with new OVMFs.

I think this is what needed to support it:

1. Add the new section type to the `SectionType` enum definition in `ovmf.py`
2. Add another if clause in function `snp_update_section` in `guest.py`; it seems that for this section it should be measured as a zero page.

**Comment by ShannonSD** [comment]
Hi @dubek,
Thank you for the response, I've modified the snp_update_section as you've explained to measure the OVMF_SECTION_TYPE_SVSM_CAA page as zero pages.

However,  the launch measurement in the attestation report does not match the digest produced by this tool.
I am uncertain whether this is because of the change in the snp_update_section or whether I am doing something wrong when trying to generate the digest.

I am using the AMDSEV repository (snp-latest branch) to create the OVMF binary, guest and host kernels (6.9.0 patched), and qemu. The build has been modified to build "OvmfPkg/AmdSev/AmdSevX64.dsc" as I'd like to measure the kernel.

The VM is launched using the launch-qemu.sh script, the launch command is:
```
/home/ubuntu/AMDSEV/snp-release-2024-05-27/usr/local/bin/qemu-system-x86_64 -enable-kvm -cpu EPYC-v4 -machine q35 -smp 4,maxcpus=255 -m 2048M,slots=5,maxmem=10240M -no-reboot -bios /home/ubuntu/AMDSEV/snp-release-2024-05-27/usr/local/share/qemu/OVMF.fd -netdev user,id=vmnic,hostfwd=tcp::8000-:22 -device e1000,netdev=vmnic,romfile= -drive file=/home/ubuntu/img/disk.qcow2,if=none,id=disk0,format=qcow2 -drive if=virtio,format=raw,file=../seed.img -device virtio-scsi-pci,id=scsi0,disable-legacy=on,iommu_platform=true -device scsi-hd,drive=disk0 -machine memory-encryption=sev0,vmport=off -object memory-backend-memfd,id=ram1,size=2048M,share=true,prealloc=false -machine memory-backend=ram1 -object sev-snp-guest,id=sev0,cbitpos=51,reduced-phys-bits=1 -nographic -monitor pty -monitor unix:monitor,server,nowait 
```

The measurement is generated using:
```
sev-snp-measure.py \
    --mode snp \
    --vmm-type=QEMU \
    --vcpus=4 \
    --vcpu-type=EPYC-v4 \
    --ovmf=AMDSEV/snp-release-2024-05-27/usr/local/share/qemu/OVMF.fd
```

I've tried using the tool within the guest VM to directly measure the kernel as well using (copied the ovmf binary into the guest):
```
sev-snp-measure.py \
    --mode snp \
    --vmm-type=QEMU \
    --vcpus=4 \
    --vcpu-type=EPYC-v4 \
    --ovmf=OVMF.fd \
    --kernel=/boot/vmlinuz
```
The measurement still doesn't match.
Finally, I've even tried using the --guest-features parameter with 1 and 33.

Do you see an issue with my use of the measurement tool?

**Comment by dubek** [comment]
Are you working with sev-snp-measure 0.0.10 or 0.0.9? On last Friday I released 0.0.10 which includes the merged PR #48 which modifies the VMSA. However, depending on the host kernel, this might not be what you want... You can try with the other version.

Also, don't pass `--kernel` to sev-snp-measure if you didn't also pass `-kernel` to QEMU and enabled `kernel-hashes=on`.

**Comment by ShannonSD** [comment]
The issue is resolved with release 0.0.10.
Thank you @dubek !

**Comment by dubek** [comment]
Thanks to @DGonzalezVillal and @larrydewey , the original issue (`ValueError: 4 is not a valid SectionType`) is fixed in PR #53 and now released as sev-snp-measure 0.0.11 (on pypi).
